### PR TITLE
refactor(core): move internal functionality to separated index

### PR DESCRIPTION
**Pull request by barak007**
separate the function and import it directly from dist

fix: #2084

### CODE REVIEW

File: packages/core/src/internal.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export * from './index';
```
Comment (tzachbon) [inline packages/core/src/internal.ts:1]: I'm not sure if it should export all of the core too.

File: packages/core/src/internal.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+export * from './internal/process-declaration-functions';
```
Comment (tzachbon) [inline packages/core/src/internal.ts:2]: Let's keep the explicit exports
```suggestion
export { processDeclarationFunctions } from './internal/process-declaration-functions';
```

**Review by tzachbon — COMMENTED** [review]
In general, it looks good.
I left a couple of questions

File: packages/core/src/internal.ts (hunk 1 above)
Comment (idoros) [reply packages/core/src/internal.ts:1]: I agree

Comment (barak007) [reply packages/core/src/internal.ts:1]: me too

Comment (barak007) [reply packages/core/src/internal.ts:1]: @idoros should we?

Comment (barak007) [reply packages/core/src/internal.ts:1]: maybe we can skip the internal.ts and just import files directly from the internal folder

Comment (idoros) [reply packages/core/src/internal.ts:1]: I think it unifies the internal api in a way that can be simply mapped to package json exports when ts 4.5 will be out

**Pull request merged**